lahja/asyncio/endpoint.py

```diff
@@ -851,7 +811,7 @@ def _remove_sync_subscription(
         A handler is passed as a second argument an :class:`~lahja.common.Subscription` is returned
         to unsubscribe from the event if needed.
         """
-        if inspect.iscoroutine(handler):
+        if inspect.iscoroutinefunction(handler):
```

Comment:
[The docs](https://docs.python.org/3/library/asyncio-task.html#asyncio.iscoroutinefunction) say that `asyncio.iscoroutinefunction()` covers one extra case which I think is the legacy way coroutines used to be implemented? It sounds slightly better to use the asyncio version just in case users are still writing coroutines the old way.